salt/loader/__init__.py

```diff
@@ -1060,8 +1060,9 @@ def grains(opts, force_refresh=False, proxy=None, context=None, loaded_base_name
         __grains__ = salt.loader.grains(__opts__)
         print __grains__['id']
     """
-    # Need to re-import salt.config, somehow it got lost when a minion is starting
-    import salt.config
+    # Without this global declaration, the late `import` statement below causes
+    # `salt` to become a function-local variable.
+    global salt
```

Comment:
Let's not make salt a global.
The import that was in place was working.

If we really wanted to fix this import, we'd have to track down the circular import issue, and most likely, it's not an easy circular import to fix.

Reply:
`salt` is already a global, just not in this function (without this change).